tools/profiling/memory/memory_diff.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import csv
+import glob
+import math
+import multiprocessing
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+import sys
+
+sys.path.append(
+    os.path.join(os.path.dirname(sys.argv[0]), '..', '..', 'run_tests',
+                 'python_utils'))
+import check_on_pr
+
+argp = argparse.ArgumentParser(description='Perform diff on microbenchmarks')
+
+argp.add_argument('-d',
+                  '--diff_base',
+                  type=str,
+                  help='Commit or branch to compare the current one to')
+
+argp.add_argument('-j', '--jobs', type=int, default=multiprocessing.cpu_count())
+
+args = argp.parse_args()
+
+_INTERESTING = {
+    'client call':
+        (rb'client call memory usage: ([0-9\.]+) bytes per call', float),
+    'server call':
+        (rb'server call memory usage: ([0-9\.]+) bytes per call', float),
+}
+
+
+def _run():
+    """Build with Bazel, then run, and extract interesting lines from the output."""
+    subprocess.check_call([
+        'tools/bazel', 'build', '-c', 'opt',
```

Comment:
nit: the bloat_diff.py uses cmake and runs under a docker image that wasn't meant to run bazel build.

So using bazel for memory_diff is fine, but probably doesn't make sense to use the same kokoro job as bloat_diff in that case?

Reply:
I don't think I'm qualified to judge there: happy to split it out into a different job, but I don't really know how to do that (and I'm a bit time starved to learn right now)

Reply:
Basically clone and modify these files: 
tools/internal_ci/linux/grpc_bloat_diff_in_docker.sh -> tools/internal_ci/linux/grpc_memory_diff_in_docker.sh
tools/internal_ci/linux/grpc_bloat_diff.sh -> tools/internal_ci/linux/grpc_memory_diff.sh

I can then help with fine tuning stuff like choosing the right docker image and with creation of the kokoro job.

Reply:
done